src/Hng.Application/Services/UserService.cs

```diff
@@ -37,5 +53,78 @@ public async Task<IEnumerable<UserDto>> GetAllUsersAsync()
             return _mapper.Map<IEnumerable<UserDto>>(users);
             
         }
+
+
+
+
+
+
+
+
+
+        public async Task<(bool IsSuccess, string ErrorMessage)> IsEmailUniqueAsync(string email)
+        {
+            var existingUser = await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
+            return existingUser == null
+                ? (true, string.Empty)
+                : (false, "Email already exists");
+        }
+
+        public async Task<UserResponseDto> CreateUserAsync(UserSignupDto userSignupDto)
+        {
```

Comment:
What is this Atrocity?

Use Automaper to map your Dto to entities and vice versa 

Create user already exists in user repository.